packages/orchestration/src/exos/agoric-names-tools.js

```diff
@@ -0,0 +1,66 @@
+import { VowShape } from '@agoric/vow';
+import { E, Far } from '@endo/far';
+import { M } from '@endo/patterns';
+import { BrandShape } from '@agoric/ertp';
+
+const { Fail } = assert;
+
+/**
+ * @import {NameHub} from '@agoric/vats';
+ * @import {AssetInfo} from '@agoric/vats/src/vat-bank';
+ * @import {Remote} from '@agoric/internal';
+ * @import {Vow, VowTools} from '@agoric/vow';
+ * @import {Zone} from '@agoric/zone';
+ */
+
+/**
+ * Perform remote calls to agoricNames in membrane-friendly way. This is an
+ * interim approach until https://github.com/Agoric/agoric-sdk/issues/9541,
+ * https://github.com/Agoric/agoric-sdk/pull/9322, or
+ * https://github.com/Agoric/agoric-sdk/pull/9519
+ *
+ * XXX consider exposing `has`, `entries`, `keys`, `values` from `NameHub`
+ *
+ * @param {Zone} zone
+ * @param {{ agoricNames: Remote<NameHub>; vowTools: VowTools }} powers
+ */
+export const makeAgNamesTools = (
```

Comment:
Reviewers might suggest a better name

Reply:
I think we should avoid `Ag` abbreviation in code. "Tools" isn't quite right because this is augmenting AgoricNames. We sometimes use Kit for that but more often that's for a multi-faceted object and this isn't.

`AgoricNames` would be confused with the remote/vat and `AgoricNameHub` could also be.

`ResumableAgoricNames` conveys that it's wrapping the remove in vows, but that seems like an anti-pattern. We don't want to have to define a fetcher for each remote.

My suggestion is that instead we name it conspicuously not for production. E.g. `AgoricNamesHack`

Reply:
I've renamed it to `ResumableAgoricNamesHack`. Given it supports additional functionality - a vbank lookup by brand with local caching - we might consider renaming in the future. When we build `getBrandInfo` and `asAmount` this may be a good place to put them (or, we can move `findBrandInVBank` there) cc @dckc.